Compiling with WITH_HTTP=1 and WITH_MQTT=0 fails

This is what #249 tries to fix, but it does not fix the actual issue.

In the argument parsing in the main function, the cases for 20 and 21 (which are psk and identity and belong to MQTT) are part of the WITH_HTTP block and not the WITH_MQTT block. This should also mean that, when compiling with MQTT but without HTTP, these settings are missing, although I have not checked this.

The full error message is:
```recorder.c: In function 'main':
recorder.c:1447:7: error: 'struct udata' has no member named 'psk'
     ud->psk = strdup(optarg);
       ^~
recorder.c:1450:7: error: 'struct udata' has no member named 'identity'
     ud->identity = strdup(optarg);
       ^~
make: *** [<builtin>: recorder.o] Error 1
```

--- a/recorder.c
+++ b/recorder.c
@@ -1417,6 +1417,12 @@ int main(int argc, char **argv)
 			case 'p':
 				ud->port = atoi(optarg);
 				break;
+			case 20:
+				ud->psk = strdup(optarg);
+				break;
+			case 21:
+				ud->identity = strdup(optarg);
+				break;
 #endif /* WITH_MQTT */
 			case 5:
 				geohash_setprec(atoi(optarg));
@@ -1442,13 +1448,6 @@ int main(int argc, char **argv)
 			case 15:
 				if (ud->browser_apikey) free(ud->browser_apikey);
 				ud->browser_apikey = strdup(optarg);
-				break;
-
-			case 20:
-				ud->psk = strdup(optarg);
-				break;
-			case 21:
-				ud->identity = strdup(optarg);
 				break;
 #endif
 			case 'D':

